.konflux/overlay/overlay.bash

```diff
@@ -0,0 +1,356 @@
+            --help)
+                usage
+                exit
+                ;;
+            --set-csv-file)
+                ARG_CSV_FILE=$2
+                shift 2
+                ;;
+            --set-pinning-file)
+                ARG_PINNING_FILE=$2
+                shift 2
+                ;;
+            --set-mapping-file)
+                ARG_MAPPING_FILE=$2
+                shift 2
+                ;;
+            --set-mapping-staging)
+                map_staging=1
+                ARG_MAP=$MAP_STAGING
+                shift 1
+                ;;
+            --set-mapping-production)
+                map_production=1
+                ARG_MAP=$MAP_PRODUCTION
+                shift 1
+                ;;
+            --)
+                shift
+                break
+                ;;
+            *)
+                echo "Error: unexpected option: $1" >&2
+                usage
+                exit 1
+                ;;
+        esac
+    done
+
+    # validate images file
+    if [[ -n $ARG_PINNING_FILE && ! -f "$ARG_PINNING_FILE" ]]; then
+        echo "Error: file '$ARG_PINNING_FILE' does not exist." >&2
+        exit 1
+    fi
+
+    # validate csv file
+    if [[ -n $ARG_CSV_FILE && ! -f "$ARG_CSV_FILE" ]]; then
+        echo "Error: file '$ARG_CSV_FILE' does not exist." >&2
+        exit 1
+    fi
+
+    # validate map options
+    if [[ $map_staging -eq 1 && $map_production -eq 1 ]]; then
+        echo "Error: cannot specify both '--set-mapping-staging' and '--set-mapping-production'." >&2
+        exit 1
+    fi
+
+    if [[ $map_staging -eq 1 || $map_production -eq 1 ]]; then
+        if [[ ! -n $ARG_MAPPING_FILE ]]; then
+            echo "Error: specify '--set-mapping-file' to use a container registry map file." >&2
+            exit 1
+        fi
+
+        if [[ ! -f "$ARG_MAPPING_FILE" ]]; then
+            echo "Error: file '$ARG_MAPPING_FILE' does not exist." >&2
+            exit 1
+        fi
+    fi
+
+    if [[ -n $ARG_MAPPING_FILE ]]; then
+        if [[ $map_staging -eq 0 && $map_production -eq 0 ]]; then
+            echo "Error: specify '--set-mapping-staging' or '--set-mapping-production'." >&2
+            exit 1
+        fi
+    fi
+
+    echo "Parsing args completed!"
+}
+
+overlay_release()
+{
+    echo "Overlaying relase..."
+
+    local display_name="lifecycle-agent-operator"
+    local description="lifecycle-agent-operator"
+    local version="4.20.0"
+    local name="lifecycle-agent-operator"
+    local name_version="$name.v$version"
+    local manager="lifecycle-agent-operator"
+    local skip_range=">=4.9.0 <4.20.0"
+    local replaces="lifecycle-agent-operator.v4.20.0"
+    # min_kube_version should match ocp
+    export min_kube_version="1.32.0"
+
+    yq e -i ".metadata.annotations[\"containerImage\"] = \"${IMAGE_TO_TARGET[$MANAGER_KEY]}\"" $ARG_CSV_FILE
+    yq e -i ".spec.displayName = \"$display_name\"" $ARG_CSV_FILE
+    yq e -i ".spec.description = \"$description\""  $ARG_CSV_FILE
```

Comment:
I did not see this getting injected in the old render template method. is this needed here ? do we want to keep the csv format as close as we can to the previous ones ?

Reply:
I believe this is needed to set various fields that we currently set in the existing midstream and was missing from the original konflux upstream implementation